scripts/build/build_examples.py

```diff
@@ -116,9 +116,14 @@ def ValidateRepoPath(context, parameter, value):
     default=False,
     is_flag=True,
     help='Skip timestaps in log output')
+@click.option(
+    '--custom-build-arg',
```

Comment:
what is the custom build arg for?

The current build system ensures that out/foo is always consistently using the same build options, we tried to not provide custom arguments even if convenient (e.g. for asan/tsan/no-ble etc)

Reply:
We quite often use custom build flags to enable optional features with `-DCONFIG_SOME_FLAG=y` arguments. If `build_examples.py` is meant to be used by devs and not just CI, it may be useful to pass additional build flags. Why is it discouraged?

Reply:
could you provide some examples? I assumed this was for things like `asan=true` or `disable_ipv4` or similar, in which cases I would prefer a separate build variant.

I would like to understand if these are really one-off (I imagine things like `-DSTACK_SIZE=1234` maybe is too custom to create build variants) and if we could/should somehow integrate into the build output naming